ouroboros-network-api/src/Ouroboros/Network/PeerSelection/RelayAccessPoint.hs

```diff
@@ -53,6 +57,49 @@ data RelayAccessPoint = RelayAccessDomain  !DNS.Domain !Socket.PortNumber
                       | RelayAccessAddress !IP.IP      !Socket.PortNumber
   deriving (Eq, Ord)
 
+-- | These instances are used to serialize 'LedgerPeerSnapshot'
+-- consensus LocalStateQuery server which uses these instances
+-- for all its query responses. It appears they provide some improved
+-- debugging diagnostics over Serialize instances.
+instance ToCBOR RelayAccessPoint where
+  toCBOR = \case
+    RelayAccessDomain domain port ->
+         encodeListLen 3
+      <> encodeWord8 0
+      <> serialise' port
+      <> toCBOR domain
+    RelayAccessAddress (IP.IPv4 ipv4) port ->
+         encodeListLen 3
+      <> encodeWord8 1
+      <> serialise' port
+      <> toCBOR (IP.fromIPv4 ipv4)
+    RelayAccessAddress (IP.IPv6 ip6) port ->
+         encodeListLen 3
+      <> encodeWord8 2
+      <> serialise' port
+      <> toCBOR (IP.fromIPv6 ip6)
+    where
+      serialise' = toCBOR . toInteger
+
+instance FromCBOR RelayAccessPoint where
+  fromCBOR = do
+    listLen <- decodeListLen
+    when (listLen /= 3) . fail $    "Unrecognized RelayAccessPoint list length "
+                                 <> show listLen
+    constructorTag <- decodeWord8
+    port <- fromInteger <$> fromCBOR @Integer
+    case constructorTag of
+      0 -> do
+        domain <- fromCBOR
+        return $ RelayAccessDomain domain port
+      1 -> do
+        ip4 <- IP.IPv4 . IP.toIPv4 <$> fromCBOR
+        return $ RelayAccessAddress ip4 port
+      2 -> do
+        ip6 <- IP.IPv6 . IP.toIPv6 <$> fromCBOR
+        return $ RelayAccessAddress ip6 port
+      _ -> fail $ "Unrecognized RelayAccessPoint tag: " <> show constructorTag
```

Comment:
Is this according to the current CDDL spec?

Reply:
Also CBOR codecs in ouroboros-network-protocols where the cddl spec is defined and the codecs are defined use:

```
import Codec.CBOR.Decoding qualified as CBOR
import Codec.CBOR.Encoding qualified as CBOR
```

and the `Codec` type. Take a look at how things are done there. I am not sure about defining the CDDL, though because this project doesn't deal with it... hmm

Reply:
I haven't seen the CDDL spec for this, but I could create one if this is a requirement. I guess we should establish who should own this spec in the first place, us, node or consensus?

Reply:
The path to writing a spec is opened now that all serialisation instances are hand written.